tests/problems.py

```diff
@@ -16,7 +16,7 @@
 import torch
 from torch import nn
 
-from torchsde import SDEIto
+from torchsde import SDEIto, BaseSDE
 
 
 class Ex1(SDEIto):
```

Comment:
Why not do the super class change for the other Ex problems just as Ex2?

Reply:
Sure! Done. 
I've added `sde_type='ito'` default arg in constructor so to not modify all `diagnostics` files. Can it stay that way?

Reply:
Cool thanks. Could you make sure all other files in `diagnostics` still run?

Reply:
I've just checked all five and four of them run but `ito_scalar` is failing with:
```
ValueError: Drift and diffusion shapes do not match. Got drift shape (32, 1), meaning (32,) batch dimensions and 1 channel dimensions, but diffusion shape (32, 1), meaning () batch dimensions, 32 channel dimensions and 1 noise dimension.
```

Reply:
Yup, that's a known bug. You can leave `ito_scalar` alone for now. (Or fix it, if you prefer.)

Reply:
Sure I can fix it, but maybe I can address it in separate PR. 

So are the drift and diffusion incorrectly defined for this diagnostic example or is this check_contract condition incorrect?

Reply:
A separate PR would be sensible, you're right.
I'd treat the example as the one that is incorrect.

Reply:
@patrick-kidger Hi! 
I've just tried to fix it but I found that for current `dev` the `python3 -m diagnostics.ito_diagonal` is also failing with:
```
TypeError: 'dict' object is not callable
```

What do you think about first setting up CI (#17) to have some confidence about project condition and then try to fix those errors? 
If it's not started I can try it as I set up some CI at work. I would see it as two checks:
1. `pytest` stage for all test suites in `tests` directory
2. `diagnostic` stage where we run files from `diagnostic` folder without plotting and only doing assertion on slope to check if it equals expected one (with some margin) so that we know if API not fails and whether methods have correct slope

Reply:
Haha, you can blame Chen for that one - it's a typo in something that just got merged in; nothing major thankfully. I'm going to push a branch / open a PR shortly, so I'll fix that at the same time.

Setting up CI sounds like an excellent idea. @lxuechen what are your plans for this? Personally I'm a big fan of GitHub Actions.

I'd suggest probably handling it as just one group of checks; the appropriate behaviour from diagonistics should instead be copied across into tests. I think the diagnostics folder is meant to be more of an informal human-readable summary than anything else.

Reply:
@mtsokol For context btw, the reason CI hasn't happened yet is that some of the tests need updating, so our checking before merging has been a bit more informal.

Reply:
Yeah, on top of fixing broken tests, I'd like to note that this is officially a google repo, so there are things that I can and cannot do. I was procrastinating on this, since I need to check for and be cautious about certain things. 

I'm happy to give this a shot later this week.